src/UglyToad.PdfPig/Writer/PdfDocumentBuilder.cs

```diff
@@ -152,21 +193,15 @@ public AddedFont AddStandard14Font(Standard14Font type)
 
             var id = Guid.NewGuid();
             var name = NameToken.Create($"F{fontId++}");
-            var added = new AddedFont(id, name);
+            var added = new AddedFont(id, context.ReserveObjectNumber());
             fonts[id] = new FontStored(added, new Standard14WritingFont(Standard14.GetAdobeFontMetrics(type)));
-
             return added;
         }
 
-        internal IndirectReference AddImage(DictionaryToken dictionary, byte[] bytes)
+        internal IndirectReferenceToken AddImage(DictionaryToken dictionary, byte[] bytes)
         {
-            var reserved = context.ReserveNumber();
-
-            var stored = new ImageStored(dictionary, bytes, reserved);
-
-            images[stored.Id] = stored;
-
-            return new IndirectReference(reserved, 0);
+            var streamToken = new StreamToken(dictionary, bytes);
```

Comment:
is `images` useless now ? or should it still hold the stored images

Reply:
Yea, they aren't needed on the PdfDocumentBuilder anymore, will clean this up.